web/src/app/alerts/CreateAlertDialog/CreateAlertDialog.tsx

```diff
@@ -40,14 +41,15 @@ const useStyles = makeStyles((theme: Theme) => ({
 
 export default function CreateAlertDialog(props: {
   onClose: () => void
-  serviceID: string
 }): JSX.Element {
   const classes = useStyles()
   const [step, setStep] = useState(0)
+  const serviceID = useLocation()[0].split('/')[2] // if exists in URL
```

Comment:
This seems a little hacky and fragile, it really should inherit the `serviceID` from the AlertsList instead of making assumptions about the URL.

What about adding some sort of helper function to return a HOC for the one-off case?

```ts
CreateDialog={withServiceID(CreateAlertDialog, props.serviceID)}
```


Or have an optional `renderCreateDialog` prop for the one-off case:

```ts
renderCreateDialog={(props)=>(
  <CreateAlertDialog serviceID={props.serviceID} {props...} />
)}
```